### PR TITLE
[instrument_builder] Blank line fix

### LINKED ISSUE #4604: [instrument_builder] Formatting Blank Line is not inserting
Adding a Blank Line is not inserting

### LINKED ISSUE #5486: [instrument builder] Add "Blank Line" does not work
Test plan Item 1
1.j Add Blank Line

In instrument builder, nothing changes when you add a "Blank Line" field. Keeping adding field, there is no evidence showing the blank line.

If you download the linst file, there is no blank line in the file.

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

See #https://github.com/aces/Loris/issues/5486

#### Testing instructions (if applicable)

1. Checkout PR.
2. Replicate the https://github.com/aces/Loris/issues/5486 gif that @coreyAbout made.

#### Links to related tickets (GitHub, Redmine, ...)

* https://github.com/aces/Loris/issues/5486
* Resolves #4604
* Resolves #5486

### CODE REVIEW

**Comment by coreyAbout** [comment]
Switched to your branch, run make dev, and not able to login: error:
`This page isn’t working cobra-lab-dev.loris.ca is currently unable to handle this request.
HTTP ERROR 500`

File: modules/instrument_builder/jsx/react.questions.js [hunk 1]
```diff
@@ -502,6 +502,10 @@ class ListElements extends Component {
     let textSize = 'small';
     // Set the options for the desired type
     switch (newId) {
+      case 'line':
+        newState.Options = {};
+        newState.Name = '';
+        newState.Description = '';
```
Comment (driusan) [inline modules/instrument_builder/jsx/react.questions.js]: There was a "blank line" option, why does this now need a linst spec change to work?

**Comment by maltheism** [comment]
@coreyAbout I'm not sure why you're getting that error. I think it's unrelated to the PR because only a few lines of the code are changed in the instrument_builder of this PR.

File: modules/instrument_builder/jsx/react.questions.js (hunk 1 above)
Comment (maltheism) [reply modules/instrument_builder/jsx/react.questions.js]: @driusan when debugging the bug I encountered that the previous state of Options, Name, Description were getting carried over when creating a 'break line' and setting them to default values would allow the 'break line' to work.

Comment (driusan) [reply modules/instrument_builder/jsx/react.questions.js]: So does this change anything in the saved file? Also, does this case need a `break`?

Comment (maltheism) [reply modules/instrument_builder/jsx/react.questions.js]: @driusan forgot to test that. I'll check and report back.

Comment (maltheism) [reply modules/instrument_builder/jsx/react.questions.js]: @driusan seems to not being adding to .linst file. I'll debug some more and fix. As of now just the frontend error is fixed for rendering the break lines.

Comment (driusan) [reply modules/instrument_builder/jsx/react.questions.js]: What about the `break` statement part? Right now this is falling through to the textarea/textbox cases (though I doubt that's the cause of it not saving properly..)

Comment (maltheism) [reply modules/instrument_builder/jsx/react.questions.js]: @driusan nice catch on the missing break. That part of the code makes the Blank Line render on the frontend of the instrument builder. Otherwise before the code change, it wasn't rendering the Blank Line to the frontend. The new code I just added fixes the save to a .linst file.

File: modules/instrument_builder/js/instrument_builder.instrument.js [hunk 2]
```diff
@@ -89,6 +89,8 @@ var Instrument = {
             }
             for (element of elements[i].Elements) {
                 switch (element.Type) {
+                    case "line":
+                        content += '<br>\n';
```
Comment (driusan) [inline modules/instrument_builder/js/instrument_builder.instrument.js]: The format for static content (including br tags) needs to be like the format of "label", including the static type and (empty) fieldname. I don't think this will load properly in LORIS as written. It needs to be

`static{@}{@}<br>\n`

File: modules/instrument_builder/jsx/react.questions.js (hunk 1 above)
Comment (christinerogers) [reply modules/instrument_builder/jsx/react.questions.js]: nice catches.  Do we also need a change to an existing linst spec file, @driusan ?  if so, let's tag update or tag as `needs documentation`

Comment (christinerogers) [reply modules/instrument_builder/jsx/react.questions.js]: Can you confirm @maltheism @coreyAbout  - 
Does this reset (wipe) the last-used Options array?  
It should keep the last options for the next (e.g. dropdown) question added, if at all possible. 
The intended behaviour is that when the user adds `dropdown1:options{a, b, c}` then `blank line` then `dropdown2` -- the options will already be loaded with `{a, b, c}`

**Comment by driusan** [comment]
@kongtiaowang can you test this (including making sure the instrument saved can be loaded by LORIS and renders the blank line)?

**Comment by maltheism** [comment]
@christinerogers I just tested the PR with dropdowns and everything seems to work.

**Comment by kongtiaowang** [comment]
load: function (file, callback) is missing the "blank line" element.

**Comment by maltheism** [comment]
@kongtiaowang please review again and thanks for the help!

File: modules/instrument_builder/jsx/react.questions.js [hunk 3]
```diff
@@ -37,6 +37,8 @@ class LorisElement extends Component {
       case 'label':
         elementHtml = <p>{element.Description}</p>;
         break;
+      case 'line':
+        elementHtml = <div></div>;
```
Comment (kongtiaowang) [inline modules/instrument_builder/jsx/react.questions.js]: missing break, so it has a tiny render issue.

Comment (maltheism) [reply modules/instrument_builder/jsx/react.questions.js]: @kongtiaowang thanks, I added it.

File: modules/instrument_builder/js/instrument_builder.instrument.js [hunk 4]
```diff
@@ -89,6 +89,9 @@ var Instrument = {
             }
             for (element of elements[i].Elements) {
                 switch (element.Type) {
+                    case "line":
```
Comment (driusan) [inline modules/instrument_builder/js/instrument_builder.instrument.js]: there is no "line" type saved in a linst file. Why did you add this type? It's always been a static label with a <br> content.

Comment (maltheism) [reply modules/instrument_builder/js/instrument_builder.instrument.js]: @driusan I removed it. I've tested the code and that part is unnecessary. I added it because I thought it was necessary from reviewing @kongtiaowang's past PR on this issue.

**Comment by kongtiaowang** [comment]
@driusan @maltheism 
After the change, when the user load this linst file again, It will show the label tag with title <br> on the page instead of a blank line.

**Comment by maltheism** [comment]
@driusan PR is ready for merge I think.

**Comment by christinerogers** [comment]
@coreyAbout  Could you re-test Monday?  this looks very ready to merge. 
Thanks @maltheism  and @kongtiaowang !

**Comment by coreyAbout** [comment]
OK @christinerogers

**Comment by coreyAbout** [comment]
Looks like more work need to be done. blank line not showing up after instrument install.
@christinerogers @driusan 

Screenshot from 2019-11-25 13_58_53
Screenshot from 2019-11-25 13_56_25

**Comment by maltheism** [comment]
@coreyAbout I think that issue isn't in the instrument_builder

**Comment by coreyAbout** [comment]
@maltheism you might be right. Wait for @driusan to indicate whether to put a new issue or not.

**Comment by maltheism** [comment]
@driusan can this PR get merged. I think the issue @coreyAbout mentioned should be solved in another PR because the issue is outside the instrument_builder module code.

**Comment by driusan** [comment]
Is the feature working in the frontend for previously built instruments that have a blank line?

**Comment by maltheism** [comment]
@driusan I'm not sure if any previous built instruments had the blank line added to the linst file. Since the code never seemed to have had the functionality for adding it to the linst files.

**Comment by driusan** [comment]
There are many in the IBIS repo if you search for `"static{@}{@}<br />"` (with quotes).. if you don't have access to the repo, maybe @sruthymathew123 or @nicolasbrossard can get you access to one.

**Comment by nicolasbrossard** [comment]
@maltheism Would you like me to send you a linst and/or class file containing an example of this?

**Comment by maltheism** [comment]
@nicolasbrossard yes please

**Comment by maltheism** [comment]
@nicolasbrossard @driusan The old linst files don't seem to work with this PR. The result ends up being a label with <br> inside it.

**Comment by zaliqarosli** [comment]
@maltheism any chance you can move this to 23.0-release branch? if that's okay with @driusan . this PR is needed for me to pass my testing of the instrument builder

**Comment by driusan** [comment]
This PR isn't going in to the 23 release. We aren't changing the LINST spec.

**Comment by zaliqarosli** [comment]
@driusan even if it's a bug fix for failed testing?

**Comment by driusan** [comment]
this is not a bug fix, this is a complete change of behaviour.

**Comment by driusan** [comment]
Replaced by #6859
